release/semver/semver.go

```diff
@@ -0,0 +1,195 @@
+package semver
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/rancher/ecm-distro-tools/types"
+
+	"golang.org/x/mod/semver"
+)
+
+type Pattern struct {
+	Pattern    string
+	Major      *int
+	Minor      *int
+	Patch      *int
+	Prerelease string
+	Build      string
+}
+
+type Version struct {
+	Version    string
+	Major      int
+	Minor      int
+	Patch      int
+	Prerelease string
```

Comment:
Can we string the hyphen from this field?

Reply:
Hyphens no longer displayed